lease/leasehttp/http.go

```diff
@@ -111,3 +148,42 @@ func RenewHTTP(id lease.LeaseID, url string, rt http.RoundTripper, timeout time.
 	}
 	return lresp.TTL, nil
 }
+
+// TimeToLiveHTTP probes the given lease ID.
+func TimeToLiveHTTP(id lease.LeaseID, url string, rt http.RoundTripper, timeout time.Duration) (*pb.LeaseTimeToLiveResponse, error) {
```

Comment:
accept context in this func? remove timeout. use context.Deadline to set the current timeout

if the context times out, cancel http request